pkg/trace/api/profiles.go

```diff
@@ -22,48 +25,78 @@ const (
 	profilingURLDefault = "https://intake.profile.datadoghq.com/v1/input"
 )
 
-// profilingEndpoint returns the profiling intake API URL based on agent configuration.
-func profilingEndpoint() string {
+// profilingEndpoints returns the profiling intake urls and their corresponding
+// api keys based on agent configuration. The main endpoint is always returned as
+// the first element in the slice. When multiple endpoints are in use, the
+// response from the main endpoint is proxied back to the client, while for
+// all aditional endpoints the response is discarded. There is no de-duplication
+// done between endpoint hosts or api keys.
+func profilingEndpoints(apiKey string) []*traceconfig.Endpoint {
+	endpoints := []*traceconfig.Endpoint{}
+	var main string
 	if v := config.Datadog.GetString("apm_config.profiling_dd_url"); v != "" {
-		return v
+		main = v
+	} else if site := config.Datadog.GetString("site"); site != "" {
+		main = fmt.Sprintf(profilingURLTemplate, site)
+	} else {
+		main = profilingURLDefault
 	}
-	if site := config.Datadog.GetString("site"); site != "" {
-		return fmt.Sprintf(profilingURLTemplate, site)
+	endpoints = append(endpoints, &traceconfig.Endpoint{Host: main, APIKey: apiKey, NoProxy: false})
+
+	if config.Datadog.IsSet("apm_config.profiling_additional_endpoints") {
```

Comment:
```suggestion
	if opt := "apm_config.profiling_additional_endpoints"; config.Datadog.IsSet(opt) {
```
Might avoid the string repetition on the two lines.